### ISSUE TITLE
plugin/acl: Not working when access with ipv6 link local address

### ISSUE THREAD

**Issue by BoringCat** [miekg/coredns#5376]
**What happened**:
Set acl to block all AAAA response. But still got response

**What you expected to happen**:
nslookup got
```
# nslookup -type=AAAA *SOMEDOMAIN* fe80::aaaa:bbbb:cccc:dddd%eth0
** server can't find *SOMEDOMAIN*: REFUSED
```

**How to reproduce it (as minimally and precisely as possible)**:
*tip: Same as Corefile below
1. listen zone `.`
2. set forward to /etc/resolv.conf
3. set acl `block type AAAA`
4. send request

**Anything else we need to know?**:
I do some test and find why this issue happened

- First, acl get client ip in here:  
  https://github.com/coredns/coredns/blob/master/plugin/acl/acl.go#L99  
  And it return by `dns.ResponseWriter.RemoteAddr().String()`  
  At  https://github.com/coredns/coredns/blob/master/request/request.go#L44-L57  
- Then, when client ip is ipv6 link local. `state.IP()` will return `Addr%ifname` instead `Addr`.   
  example: 
    - IPv6 global unicast: `2001:1111:2222:3333:4444:5555:6666:7777`  
    - IPv6 link local: `fe80::aaaa:bbbb:cccc:dddd%eth0`  
- But, `net.ParseIP` can not deal with it. Will return `<nil>`
- And at https://github.com/coredns/coredns/blob/master/plugin/acl/acl.go#L109-L112  
  when send `<nil>` to `policy.filter.GetByIP`. It return `<nil> false`. And passed `if !contained`

So acl think not need to handle this request

**Environment**:
```
Linux Router 5.4.162 #0 SMP PREEMPT Sat Dec 18 02:50:04 2021 aarch64 GNU/Linux
```

- the version of CoreDNS:
  ```
  CoreDNS-1.9.1
  linux/arm64, go1.17.8, 4b597f8
  ```
- Corefile:
  ```
  . {
      prometheus :9153
      acl {
          block type AAAA
      }
      forward . /etc/resolv.conf
  }
  ```
- logs, if applicable:  
  plugin/log is disabled.
- OS (e.g: `cat /etc/os-release`):
  ```
  NAME="OpenWrt"
  VERSION="SNAPSHOT"
  ID="openwrt"
  ID_LIKE="lede openwrt"
  PRETTY_NAME="OpenWrt SNAPSHOT"
  VERSION_ID="snapshot"
  HOME_URL="https://openwrt.org/"
  BUG_URL="https://bugs.openwrt.org/"
  SUPPORT_URL="https://forum.openwrt.org/"
  BUILD_ID="r3995-9cfbc2146"
  OPENWRT_BOARD="rockchip/armv8"
  OPENWRT_ARCH="aarch64_generic"
  OPENWRT_TAINTS="override"
  OPENWRT_DEVICE_MANUFACTURER="OpenWrt"
  OPENWRT_DEVICE_MANUFACTURER_URL="https://openwrt.org/"
  OPENWRT_DEVICE_PRODUCT="Generic"
  OPENWRT_DEVICE_REVISION="v0"
  OPENWRT_RELEASE="OpenWrt SNAPSHOT r3995-9cfbc2146"
  ```
- Others:
  nil

**Comment by BoringCat** [comment]
And here is my test code
```go
package main

import (
	"fmt"
	"net"

	"github.com/miekg/dns"
)

func main() {
	mux := dns.NewServeMux()
	mux.HandleFunc(".", func(w dns.ResponseWriter, m *dns.Msg) {
		fmt.Println(w.RemoteAddr().String())
		ip, _, _ := net.SplitHostPort(w.RemoteAddr().String())
		fmt.Println(net.ParseIP(ip))
		w.WriteMsg(m)
	})
	fmt.Println(dns.ListenAndServe(":5353", "udp", mux))
}
```

When use `dig www.google.com @fe80::aaaa:bbbb:cccc:dddd%ens3 -p 5353`. It will print
```
[fe80::aaaa:bbbb:cccc:dddd%ens3]:55795
<nil>
```

When use `dig www.google.com @::1 -p 5353`. It will print
```
[::1]:36892
::1
```

**Comment by chrisohaver** [comment]
I recall this being discussed before as a possible cause of another issue: 
* https://github.com/coredns/coredns/issues/4697#issuecomment-861504676

At a glance, it looks like an easy fix.

**Comment by SamrathPalSingh** [comment]
@chrisohaver is it enough to handle the Addr%ifname case and 'actionBlock' any request for which the parsing (net.ParseIP) fails as a rule? I can quickly raise a PR for these changes. 

Happy to include any other suggestions, in the PR, as well.

**Comment by chrisohaver** [comment]
> @chrisohaver is it enough to handle the Addr%ifname case and 'actionBlock' any request for which the parsing (net.ParseIP) fails as a rule? I can quickly raise a PR for these changes.
> 
> Happy to include any other suggestions, in the PR, as well.

@SamrathPalSingh,  perhaps I would understand better seeing the code in a PR. Thanks!

**Comment by chrisohaver** [comment]
closed by #5387
